src/cli.js

```diff
@@ -15,11 +15,57 @@ const logging = require("./logging.js");
 
 const SCHEMASTORE_CATALOG_URL =
   "https://www.schemastore.org/api/json/catalog.json";
+
+const SCHEMASTORE_CATALOG_SCHEMA_URL =
+  "https://www.schemastore.org/api/json/schema-catalog.json";
+
 const CACHE_DIR = path.join(os.tmpdir(), "flat-cache");
 
-async function getSchemaUrlForFilename(filename, cache, ttl) {
-  const { schemas } = await cachedFetch(SCHEMASTORE_CATALOG_URL, cache, ttl);
+async function getFromUrlOrFile(location, cache, ttl) {
+  return isUrl(location)
+    ? await cachedFetch(location, cache, ttl)
+    : JSON.parse(fs.readFileSync(location, "utf8").toString());
+}
+async function getSchemaUrlForFilename(catalogs, filename, cache, ttl) {
+  for (const [i, catalogLocation] of catalogs.entries()) {
+    const catalog = await getFromUrlOrFile(catalogLocation, cache, ttl);
+    const catalogSchema = await getFromUrlOrFile(
+      SCHEMASTORE_CATALOG_SCHEMA_URL,
+      cache,
+      ttl
+    );
+
+    // Validate the catalog
+    const valid = await validate(catalog, catalogSchema, cache, ttl);
+    if (!valid || catalog.schemas === undefined) {
+      throw new Error(`❌ Malformed catalog at ${catalogLocation}`);
+    }
 
+    const { schemas } = catalog;
+    const matches = getSchemaMatchesForFilename(schemas, filename);
+    if (matches.length === 1) {
+      console.log(`ℹ️ Found schema in ${catalog} ...`);
```

Comment:
```suggestion
      console.log(`ℹ️ Found schema in ${catalogLocation} ...`);
```

With `catalog` This prints `Found schema in [object Object]`

Reply:
Fixed in 7b3428af9bfd65407dfceebcb7e6bd12de6de600